packages/expo-calendar/ios/CalendarModule.swift

```diff
@@ -303,7 +330,18 @@ public class CalendarModule: Module {
     }
 
     AsyncFunction("createEventInCalendarAsync") { (event: Event, promise: Promise) in
-      try checkCalendarPermissions()
+      // Check if we have calendar access
+      if #available(iOS 17.0, *) {
+        let authStatus = EKEventStore.authorizationStatus(for: .event)
```

Comment:
I think we should reuse the permission requester mechanism – I know it's quite overcomplicated, but it would allow us to handle permissions correctly on Expo Go – right now if you give one experience write permissions all of them get these premissions